telegram/ext/_picklepersistence.py

```diff
@@ -151,7 +185,11 @@ def __init__(
     def _load_singlefile(self) -> None:
         try:
             with self.filepath.open("rb") as file:
-                data = pickle.load(file)
+                if self.bot is None:
+                    data = pickle.load(file)
+                else:
+                    data = BotUnpickler(file, self.bot).load()
```

Comment:
why not always `BotUpickler`? in case `self.bot is None`, `None` is still more helpful than `"known_bot"`, isn't it?

Reply:
I guess, just followed your [mock up ](https://nekobin.com/lifojagumo.py)

Reply:
xD